database/database.go

```diff
@@ -0,0 +1,255 @@
+package database
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/NebulousLabs/skynet-accounts/build"
+	"github.com/NebulousLabs/skynet-accounts/user"
+
+	"gitlab.com/NebulousLabs/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+var (
+	// SkynetDBHostEV holds the name of the environment variable for DB host.
+	SkynetDBHostEV = "SKYNET_DB_HOST"
+	// SkynetDBPortEV holds the name of the environment variable for DB port.
+	SkynetDBPortEV = "SKYNET_DB_PORT"
+	// SkynetDBUserEV holds the name of the environment variable for DB username.
+	SkynetDBUserEV = "SKYNET_DB_USER"
+	// SkynetDBPassEV holds the name of the environment variable for DB password.
+	SkynetDBPassEV = "SKYNET_DB_PASS"
+
+	// SkynetDBName defines the name of Skynet's database.
+	SkynetDBName = "skynet"
+	// SkynetDBUsersCollection defines the name of the "users" collection within
+	// skynet's database.
+	SkynetDBUsersCollection = "users"
+
+	// ErrUserNotFound is returned when we can't find the user in question.
+	ErrUserNotFound = errors.New("user not found")
+	// ErrEmailAlreadyUsed is returned when we try to use an email to either
+	// either create or update a user and another user already uses this email.
+	ErrEmailAlreadyUsed = errors.New("email already in use by another user")
+	// ErrNoDBConnection is returned when we can't connect to the database.
+	ErrNoDBConnection = errors.New("no connection to the database")
+
+	// True is a helper, so we can easily provide a *bool for UpdateOptions.
+	True = true
+	// False is a helper, so we can easily provide a *bool for UpdateOptions.
+	False = false
+)
+
+// DB represents a MongoDB database connection.
+type DB struct {
+	ctx   context.Context
+	db    *mongo.Database
+	users *mongo.Collection
```

Comment:
If the fields are never assigned to after init they should be static and otherwise they need a mutex.

Reply:
Makes sense.

The same should be valid for the `User` type. But there we want to lock it form the `DB` type. Would it be acceptable to expose the mutex in this case?

Reply:
Is the `User` type ever shared between threads? Does it even need a `Mutex`? Isn't every user I fetch from the DB a new copy of the current state within the DB?

Reply:
For the moment there are no threads int he service. I don't know if we'll ever need them.

Yes, the user struct is a new copy of the DB state. I'm not sure if it needs a mutex.

Reply:
F/U: #5